File: Semana_9_NotasClase_AlgoritmosDeOrdenamiento/SelectionSort_DeMenorAMayor.c
// Implementación de Selection sort
#include <stdio.h>

void swap(int *xp, int *yp)
{
    int temp = *xp;
    *xp = *yp;
    *yp = temp;
}

void selectionSort(int arr[], int n) #lo ordena de menor a mayor
{
    int i, j, min_idx;

    // Límite de movimiento uno por uno del subarreglo no ordenado
    for (i = 0; i < n-1; i++)
    {
        // Se busca el elemento mínimo en un arreglo sin ordenar
        min_idx = i;
        for (j = i+1; j < n; j++)
            if (arr[j] < arr[min_idx])
                min_idx = j;

        // Intercambiar el elemento mínimo encontrado con el primer elemento
        if(min_idx != i)
            swap(&arr[min_idx], &arr[i]);
    }
}

/* Función para imprimir */
void printArray(int arr[], int size)
{
    int i;
    for (i=0; i < size; i++)
        printf("%d ", arr[i]);
    printf("\n");
}

// Función main
int main()
{
    int arr[] = {64, 75, 2, 13, 91};
    int n = sizeof(arr)/sizeof(arr[0]);//para saber cuantos elementos hay en el arreglo, has de cuenta que 64 ocupa 4(no sé si bytes o bits), en total en el arreglo hay 20(bites o bits no sé)
    selectionSort(arr, n);
    printf("Arreglo ordenado: \n");
    printArray(arr, n);
    return 0;
}

File: Semana_9_NotasClase_AlgoritmosDeOrdenamiento/SelectionSort_DeMenorAMayor.py
#Implementación de Selection sort

def swap(lista, menor, mayor): #lista si es arr por que todo se envía por referencia, no importa que le hayas cambiado el nombre
    #me acuerdo que en C si hay swap
    print("x({})".format(menor))
    print("con y({})".format(mayor))
    temp = lista[menor] #temp no guarda índices bro
    lista[menor]= lista[mayor] #reccuerda que las listas no las modificas como un arreglo, o creo que si, ah caray, intentemoslo
    lista[mayor]= temp

def selectionSort(list, n): #minimun es el índice del mínimo
    print("Me enviaste este arreglo: {}".format(list))

    for i in range (0, n-1): #no toca n-1
        minimum = i #min_inx es integer
        print("no sé pa' que te sirva, pero minimum vale {}".format(minimum))
        for j in range (i+1, n): #no toca n
            print("no sé pa' que te sirva, pero j vale {}".format(j))
            if list[j] < list[minimum]: #las listas tienen índices
                print("Entró al famoso if con {}<{}".format(list[j], list[minimum]))
                minimum = j
                print("minimun lo cambio a {}".format(j))

        if minimum != i:
            print("Hizo swap, a continuación te lo muestro: ")
            swap(list, minimum, i) #minimum e i son enteros, guardan indices
            print("Así quedó la lista: {}".format(list))

    print("Arreglo ordenado: {}".format(list))

# Función para imprimir 
"""
def imprimirLista(arr, size):
    for i in range (0, size): #no toca size, y weno sabemos que aumenta de uno en uno
        print(arr[i])#las listas tienen índices
"""

# Función main
if __name__ == "__main__":
    lista = [64, 75, 2, 13, 91] #list son como los arreglos de Python
    n = 5
    selectionSort(lista, n)
